src/backend/cgcod.c

```diff
@@ -726,7 +726,7 @@ code *prolog()
     assert(usedalloca != 1);
     AllocaOff = alignsection(usedalloca ? (Foff - REGSIZE) : Foff, REGSIZE, bias);
 
-    CSoff = alignsection(AllocaOff - cstop * REGSIZE, REGSIZE, bias);
+    CSoff = alignsection(AllocaOff - (int)cstop * REGSIZE, REGSIZE, bias);
```

Comment:
> I'd really like to see a case where this ever happens.

Almost every function that needs a stack frame: the example above, or even simpler:
```
int foo()
{
    int[8] a;
    return a[0];
}
```
Fire up your favorite debugger or add these lines below the assignment to localsize:
```
    printf("AllocaOff = x%x, cstop = x%llx, CSoff = x%llx, NDPoff = x%llx, localsize = x%llx\n",
        AllocaOff, (long long)cstop, (long long)CSoff, (long long)NDPoff, (long long)localsize);
```
With a debug build (with dmc) of dmd and compiling for Win32 yields:
```
AllocaOff = xffffffe0, cstop = x0, CSoff = xffffffe0, NDPoff = xffffffe0, localsize = xffffffff00000020
```
while the release build produces
```
AllocaOff = xffffffe0, cstop = x0, CSoff = xffffffe0, NDPoff = xffffffe0, localsize = x20
```
The result of localsize is unexpected in the release build, considering both `NDPoff` and `localsize` are `target_size_t` aka `unsigned long long`.